File: DataBinding/DataBinding_sample/DataBindingEx1/app/src/main/java/kimjiwon/sigongeducation/databindingex1/MainActivity.java
package kimjiwon.sigongeducation.databindingex1;

import android.databinding.DataBindingUtil;
import android.databinding.ObservableArrayList;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import kimjiwon.sigongeducation.databindingex1.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // activity_main.xml에 databinding식을 작성함으로써 ActivityMainBinding클래스 자동 생성.
        // MainActivity와 activity_main을 연결.
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        // 레이아웃 상의 userList 변수에 값을 세팅해주는 작업.
        ObservableArrayList<UserData> userList = new ObservableArrayList<>();
        binding.setUserList(userList);

        // recyclerview adapter를 생성하여 레이아웃의 userdata_recyclerview에 adapter를 세팅.
        UserAdapter adapter = new UserAdapter();
        binding.userdataRecyclerview.setAdapter(adapter);

        // 더미 데이터
        userList.add(new UserData("김지원", 26, R.drawable.image1));
        userList.add(new UserData("이동수", 27, R.drawable.image2));
        userList.add(new UserData("김민수", 29, R.drawable.image3));

        Handlers handlers = new Handlers();
        binding.setHandlers(handlers);
    }
}

File: SwipeRefreshLayout/SwipeRefreshLayout_sample/SwipeRefreshLayoutEx/app/src/main/java/client/kimjiwon/swiperefreshlayoutex/MainActivity.kt
package client.kimjiwon.swiperefreshlayoutex

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.os.Handler
import android.support.v4.widget.SwipeRefreshLayout
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import client.kimjiwon.swiperefreshlayoutex.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var mBinding: ActivityMainBinding
    private var mRecyclerAdapter: RecyclerAdapter? = null
    private var userDatas: ArrayList<MainModel> = ArrayList()
    private var index: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this@MainActivity, R.layout.activity_main)
        mBinding.swipeRefreshLayout.setOnRefreshListener(onClickRefresh)


        var items: ArrayList<MainModel> = ArrayList()
        for (i: Int in 0..2) {
            items.add(MainModel("김지원", index++, R.drawable.abc_ic_star_black_48dp))
        }
        userDatas = items
        mRecyclerAdapter = RecyclerAdapter(items)
        mBinding.recyclerView.adapter = mRecyclerAdapter
        mBinding.recyclerView.layoutManager = LinearLayoutManager(this@MainActivity)
        mBinding.executePendingBindings()
    }

    private val onClickRefresh = SwipeRefreshLayout.OnRefreshListener {
        when (index % 4) {
            0 -> mBinding.swipeRefreshLayout.setColorSchemeResources(android.R.color.holo_red_light)
            1 -> mBinding.swipeRefreshLayout.setColorSchemeResources(android.R.color.holo_blue_bright)
            2 -> mBinding.swipeRefreshLayout.setColorSchemeResources(android.R.color.holo_green_light)
            3 -> mBinding.swipeRefreshLayout.setColorSchemeResources(android.R.color.holo_red_light, android.R.color.holo_blue_bright, android.R.color.holo_green_light)
        }
        
        var items: ArrayList<MainModel> = ArrayList()
//        for (i: Int in 0..3) {
        items.add(MainModel("김지원", index++, R.mipmap.ic_launcher_round))
//        }
        mRecyclerAdapter?.addItems(items)
        Handler().postDelayed(Runnable { mBinding.swipeRefreshLayout.isRefreshing = false }, 2500)
    }

}
